### PR TITLE
gate createDepositAddress emulated

### PR DESCRIPTION
**Pull request by samgermain**
```
2023-01-16T18:05:48.430Z
Node.js: v18.4.0
CCXT v2.6.25
gate.createDepositAddress (SOL)
2023-01-16T18:05:53.681Z iteration 0 passed in 835 ms

{
  info: {
    currency: 'SOL',
    address: 'CSQwbhvxfY8oE2irc55hPAvMSe4DSANvrR22dhdCZa3F',
    multichain_addresses: [
      {
        chain: 'SOL',
        address: 'CSQwbhvxfY8oE2irc55hPAvMSe4DSANvrR22dhdCZa3F',
        payment_id: '',
        payment_name: '',
        obtain_failed: '0'
      }
    ]
  },
  code: 'SOL',
  currency: 'SOL',
  address: 'CSQwbhvxfY8oE2irc55hPAvMSe4DSANvrR22dhdCZa3F',
  tag: undefined,
  network: undefined
}
2023-01-16T18:05:53.681Z iteration 1 passed in 835 ms
```

### CODE REVIEW

File: js/gate.js [hunk 1]
```diff
@@ -1547,11 +1548,25 @@ module.exports = class gate extends Exchange {
         return result;
     }
 
+    async createDepositAddress (code, params = {}) {
+        /**
+         * @method
+         * @name gate#createDepositAddress
+         * @description emulated via fetchDepositAddress, because both methods use teh same endpoint
```
Comment (sc0Vu) [inline js/gate.js:1555]: typo: `teh` => `the`

File: js/gate.js [hunk 2]
```diff
@@ -68,6 +68,7 @@ module.exports = class gate extends Exchange {
                 'borrowMargin': true,
                 'cancelAllOrders': true,
                 'cancelOrder': true,
+                'createDepositAddress': 'emulated',
```
Comment (sc0Vu) [inline js/gate.js:71]: Should we use `true` instead of `emulated`? I found we did this in bitfinex.

Comment (carlosmiei) [reply js/gate.js:71]: @sc0Vu actually I changed my opinion, the endpoint will generate the address if does not exist so this is not emulated behavior 
image

**Comment by carlosmiei** [comment]
@samgermain @ttodua @AlgoQ Note for the future: We should probably refactor `createDepositAddress` and `fetchDepositAddress` to support multiple chains (similar to `fetchDWFees`)

**Comment by carlosmiei** [comment]
LGTM ✅ 

```
 p gate createDepositAddress "BTC"
Python v3.10.8
CCXT v2.6.26
gate.createDepositAddress(BTC)
{'address': '1FHiPKKob3qbkM283gpARnQxdqquJwiY59',
 'code': 'BTC',
 'currency': 'BTC',
 'info': {'address': '1FHiPKKob3qbkM283gpARnQxdqquJwiY59',
          'currency': 'BTC',
          'multichain_addresses': [{'address': '1FHiPKKob3qbkM283gpARnQxdqquJwiY59',
                                    'chain': 'BTC',
                                    'obtain_failed': '0',
                                    'payment_id': '',
                                    'payment_name': ''},
                                   {'address': '0xC5AEBBC9071F92359D7A8c62546C4033Ec6D3996',
                                    'chain': 'BSC',
                                    'obtain_failed': '0',
                                    'payment_id': '',
                                    'payment_name': ''}]},
 'network': None,
 'tag': None}
```

**Review by carlosmiei — APPROVED** [review]

**Comment by ttodua** [comment]
> fetchDWFees

Actually, my thoughts on this was to add `networkCode` in arguments of the signature. because when generating or fetching deposit address, exchanges behave differently (some return all addresses at once, some return only one address depending the network-parameter in request). so, the common between them would be to target specific network, which would be possible with `networkCode` argument in signature, so users would get what they request.

**Comment by samgermain** [comment]
> @samgermain @ttodua @AlgoQ Note for the future: We should probably refactor `createDepositAddress` and `fetchDepositAddress` to support multiple chains (similar to `fetchDWFees`)

There is no parameter for multiple chains on the exchange, there is only the currency parameter

**Comment by carlosmiei** [comment]
> > @samgermain @ttodua @AlgoQ Note for the future: We should probably refactor `createDepositAddress` and `fetchDepositAddress` to support multiple chains (similar to `fetchDWFees`)
> 
> There is no parameter for multiple chains on the exchange, there is only the currency parameter
@samgermain, we also implement `fetchTickers` when the exchange does not accept a list of tickers in the request because we can always filter afterwards. Here is the same, if most exchanges return different addresses depending on the network, we should properly format our response.

**Comment by samgermain** [comment]
> > > @samgermain @ttodua @AlgoQ Note for the future: We should probably refactor `createDepositAddress` and `fetchDepositAddress` to support multiple chains (similar to `fetchDWFees`)
> > 
> > 
> > There is no parameter for multiple chains on the exchange, there is only the currency parameter
> > @samgermain, we also implement `fetchTickers` when the exchange does not accept a list of tickers in the request because we can always filter afterwards. Here is the same, if most exchanges return different addresses depending on the network, we should properly format our response.

I don't believe that's the case though, I think it just creates an address using the default network

**Pull request merged**